File: OOP-Java/output/com/output14/VehicleUse.java
package com.basic;

 class Vehicle {
	String color;
	private int maxSpeed;

	public int getMaxSpeed() {
		return maxSpeed;
	}

	public void setMaxSpeed(int maxSpeed) {
		this.maxSpeed = maxSpeed;
	}

	public void print() {
		System.out.println("Vehicle color : " + color);
		System.out.println("Vehicle Speed : " + maxSpeed);
	}

}

 class Car extends Vehicle {
	int numGears;
	boolean isConvertible;

	public void print() {
		System.out.println("Car color : " + color);
		System.out.println("Car Speed : " + getMaxSpeed());
		System.out.println("Car numGears : " + numGears);
		System.out.println("Car isConvertible : " + isConvertible);
	}
}

public class VehicleUse {

	public static void main(String[] args) {
		Vehicle v = new Vehicle();
		v.color = "Black";
		v.setMaxSpeed(10);
		v.print();

		Car c = new Car();
		c.numGears = 10;
		c.color = "Black";
		c.setMaxSpeed(100);
		c.print();
	}
}

File: OOP-C++/Part3/Basics/1 Inheritance/Protected Inheritance/VehicleUse.cpp
#include <iostream>
using namespace std;

#include "Car.cpp"
#include "Vehicle.cpp"

int main() {
    Vehicle v;
    v.color = "Blue";
    //v.maxSpeed = 100;		//Error: 'maxSpeed' is private  member of Vehicle.
    //v.numTyres = 4;		//Error: 'numTyres' is protected  member of Vehicle.

    Car c;
    //c.color = "Black";	//Error: 'color' become protected after inheritance
    //c.numTyres = 4;		//Error: 'numTyres' is protected within this context
    //c.maxSpeed = 220;		//Error: 'maxSpeed' is protected within this context
    c.numGears = 5;
    c.print();
}

/*
   Main has only access to non inherited public property of the class Vehicle and Car.
 */

/*
   NumTyres : -679974545
   Color :
   Num gears : 5
 */
